packages/evm/src/generic.ts

```diff
@@ -0,0 +1,254 @@
+import type { EthereumConfig } from '@buildwithsygma/core';
+import { Config, Network, ResourceType } from '@buildwithsygma/core';
+import { Bridge__factory } from '@buildwithsygma/sygma-contracts';
+import { Web3Provider } from '@ethersproject/providers';
+import type {
+  Abi,
+  AbiParametersToPrimitiveTypes,
+  ExtractAbiFunction,
+  ExtractAbiFunctionNames,
+} from 'abitype';
+import { constants, ethers } from 'ethers';
+
+import type { BaseTransferParams } from './baseTransfer.js';
+import { BaseTransfer } from './baseTransfer.js';
+import { getFeeInformation } from './fee/getFeeInformation.js';
+import type { TransactionRequest } from './types.js';
+import { createPermissionlessGenericDepositData, toHex } from './utils/helpers.js';
+import { genericMessageTransfer } from './utils/index.js';
+import { createTransactionRequest } from './utils/transaction.js';
+
+/**
+ * Required parameters for initiating
+ * a generic message transfer request
+ */
+export interface GenericMessageTransferRequest<
+  ContractAbi extends Abi,
```

Comment:
I think here we can have a separate type that we can pass here and below in the `createCrossChainContractCall`function and below

Reply:
@wainola can you elaborate a bit more?

Reply:
already discussed